src/audio_device.rs

```diff
@@ -34,7 +34,7 @@ pub async fn find_default_name(class: Class) -> Result<String> {
         .context("Failed to push future to main thread pool")?
         .await
     {
-        Ok(res) => Ok(res),
+        Ok(res) => res,
```

Comment:
The entire function body could be simplify too to the following, so we can drop `THREAD_POOL` too:

```rust
pub async fn find_default_name(class: Class) -> Result<String> {
    match gio::spawn_blocking(move || find_default_name_gst(class))
        .await
        .map_err(|err| anyhow!("Failed to spawn blocking task: {:?}", err))?
    {
        Ok(res) => Ok(res),
        Err(err) => {
            tracing::warn!("Failed to find default name using gstreamer: {:?}", err);
            tracing::debug!("Manually using libpulse instead");

            let pa_context = pa::Context::connect().await?;
            pa_context.find_default_device_name(class).await
        }
    }
}
```